### ISSUE TITLE
circular dependency in astropy.io.misc

### ISSUE THREAD

**Issue by joshuajoseph** [astropy/astropy#4750]
I want to import the `hdf5` package into my code and use the function `read_table_hdf5`, but when I do, I get this error:

after having this line of code:

`from astropy.io.misc import hdf5 as hdf`

`/Users/joshuajd/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/astropy/io/misc/hdf5.py` in <module>()
     15 from ...utils.exceptions import AstropyUserWarning
     16 from ...extern import six
`---> 17 from ...table import meta`

`/Users/joshuajd/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/astropy/table/__init__.py` in <module>()
     30 from ..io.ascii import connect
     31 from ..io.fits import connect
`---> 32 from ..io.misc import connect`

`/Users/joshuajd/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/astropy/io/misc/connect.py` in <module>()
      6                         unicode_literals)
      7 
`----> 8 from .hdf5 import read_table_hdf5, write_table_hdf5, is_hdf5`
      9 
     10 from .. import registry as io_registry
`ImportError: cannot import name read_table_hdf5`

So as you see, hdf5 depends on table, which depends on connect, which depends on hdf5... so I can't import the function I would like to!

**Comment by embray** [comment]
I saw your S/O post a second ago.  This is actually a duplicate of #1402, though via a slightly different path.  The workaround is to make sure to explicitly import `astropy.table` first.

I'm actually surprised this is still a bug. I thought @astrofrog reworked this at some point but I don't remember.
